jetbrains-core/src/software/aws/toolkits/jetbrains/services/dynamodb/editor/DynamoDbTableEditor.kt

```diff
@@ -0,0 +1,167 @@
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import software.amazon.awssdk.services.dynamodb.model.ExecuteStatementRequest
+import software.amazon.awssdk.services.dynamodb.model.KeySchemaElement
+import software.amazon.awssdk.services.dynamodb.model.KeyType
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.services.dynamodb.DynamoDbUtils.executeStatementPaginator
+import software.aws.toolkits.jetbrains.services.dynamodb.Index
+import software.aws.toolkits.jetbrains.services.dynamodb.toAttribute
+import software.aws.toolkits.jetbrains.utils.getCoroutineBgContext
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import java.awt.BorderLayout
+import java.beans.PropertyChangeListener
+import javax.swing.JComponent
+
+class DynamoDbTableEditor(private val dynamoTable: DynamoDbVirtualFile) : UserDataHolderBase(), FileEditor {
+    data class EditorState(var maxResults: Int = DEFAULT_MAX_RESULTS)
+
+    private val coroutineScope = CoroutineScope(CoroutineName("DynamoDbTableEditor") + SupervisorJob())
+    private val bg = getCoroutineBgContext(disposable = this)
+    private val edt = getCoroutineUiContext(disposable = this)
+
+    val editorState = EditorState()
+
+    private val loadingPanel = JBLoadingPanel(BorderLayout(), this)
+
+    private lateinit var searchPanel: SearchPanel
+    private val searchResults = SearchResultsPanel()
+
+    init {
+        // Async load in the editor so we can get the table info
+        loadingPanel.startLoading()
+
+        coroutineScope.launch(bg) {
+            val tableInfo = getTableInfo(dynamoTable.tableName)
+
+            launch(edt) {
+                searchPanel = SearchPanel(
+                    tableInfo = tableInfo,
+                    initialSearchType = SearchPanel.SearchType.Scan,
+                    initialSearchIndex = tableInfo.tableIndex,
+                    runAction = {
+                        executeSearch()
+                    }
+                )
+
+                loadingPanel.add(searchPanel.getComponent(), BorderLayout.NORTH)
+                loadingPanel.add(searchResults, BorderLayout.CENTER)
+
+                executeSearch(PREVIEW_SIZE)
+
+                loadingPanel.stopLoading()
+            }
+        }
+    }
+
+    private fun getTableInfo(tableName: String): TableInfo {
+        fun keySchemaToIndex(name: String?, keySchema: List<KeySchemaElement>) = Index(
+            displayName = name ?: tableName,
+            indexName = name,
+            partitionKey = keySchema.first { it.keyType() == KeyType.HASH }.attributeName(),
+            sortKey = keySchema.find { it.keyType() == KeyType.RANGE }?.attributeName()
+        )
+
+        val describeResponse = dynamoTable.dynamoDbClient.describeTable {
```

Comment:
might be a dupe comment but needs error handling (can't find my old comment and github is being weird)

Reply:
Added, heres a screen shot
<img width="604" alt="Screen Shot 2021-05-14 at 2 25 15 PM" src="https://user-images.githubusercontent.com/8992246/118333270-43413700-b4c0-11eb-8830-40265bb6f2c6.png">